### PR TITLE
Add dialog extension

### LINKED ISSUE #2662: Add specific control type for dialog
**Is your feature request related to a problem? Please describe.**
Right now, if the options are at least 10, it will automatically use dialog, but we would like to use a dialog even if the options are [less than 10](https://github.com/google/android-fhir/blob/master/datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapter.kt#L296).

**Describe the solution you'd like**
Add specific control type for dialog, that is [lookup](https://hl7.org/fhir/R4/codesystem-questionnaire-item-control.html#4.3.14.399.2)

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by FikriMilano**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #2662

**Description**
- Define dialog extension https://github.com/google/android-fhir/dialog
- Use dialog: if the item control is checkbox or radio button, and has dialog extension

FHIR Chat discussion about adding `dialog` control type to Jira:
https://chat.fhir.org/#narrow/stream/179255-questionnaire/topic/dialog.20item.20control

**Alternative(s) considered**
N/A

**Type**
Feature

**Screenshots (if applicable)**

**Checklist**

- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://google.github.io/android-fhir/contrib/) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Review by jingtang10 — CHANGES_REQUESTED** [review]
can you add a test case in `QuestionnaireEditAdapterTest.kt`?

**Comment by FikriMilano** [comment]
As discussed, we should use dialog extension instead of adding a dialog item control type

File: datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapterTest.kt [hunk 1]
```diff
@@ -496,6 +497,84 @@ class QuestionnaireEditAdapterTest {
       .isEqualTo(QuestionnaireViewHolderType.DROP_DOWN.value)
   }
 
+  @Suppress("ktlint:standard:max-line-length")
+  @Test
+  fun getItemViewType_choiceItemType_itemControlExtensionWithRadioButton_andDialogExtension_shouldReturnDialogSelectViewHolderType() {
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapterTest.kt:502]: use backtick notation for test names:
```
fun `should do something`
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapterTest.kt [hunk 2]
```diff
@@ -496,6 +497,84 @@ class QuestionnaireEditAdapterTest {
       .isEqualTo(QuestionnaireViewHolderType.DROP_DOWN.value)
   }
 
+  @Suppress("ktlint:standard:max-line-length")
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapterTest.kt:500]: you can remove this if you follow the next comment.

**Review by jingtang10 — COMMENTED** [review]
can you pls add some documentation in the docs folder (this will be automatically published on our doc site)?

File: datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireItemComponents.kt [hunk 3]
```diff
@@ -61,6 +61,9 @@ internal const val EXTENSION_ITEM_CONTROL_URL_ANDROID_FHIR =
 internal const val EXTENSION_ITEM_CONTROL_SYSTEM_ANDROID_FHIR =
   "https://github.com/google/android-fhir/questionnaire-item-control"
 
+internal const val EXTENSION_DIALOG_URL_ANDROID_FHIR =
+  "https://github.com/google/android-fhir/dialog"
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireItemComponents.kt:65]: keep extension urls in the same "namespace". see the location widget extension.

```suggestion
  "https://github.com/google/android-fhir/StructureDefinition/dialog"
```

Comment (FikriMilano) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireItemComponents.kt:65]: done

File: datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapterTest.kt (hunk 2 above)
Comment (FikriMilano) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapterTest.kt:500]: done

File: datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapterTest.kt (hunk 1 above)
Comment (FikriMilano) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapterTest.kt:502]: done

File: datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapter.kt [hunk 4]
```diff
@@ -241,8 +242,11 @@ internal class QuestionnaireEditAdapter(
   ): QuestionnaireViewHolderType {
     val questionnaireItem = questionnaireViewItem.questionnaireItem
 
-    // Use the view type that the client wants if they specified an itemControl
-    return questionnaireItem.itemControl?.viewHolderType
+    // Use the view type that the client wants if they specified an itemControl or dialog extension
+    return when {
+      questionnaireItem.shouldUseDialog -> QuestionnaireViewHolderType.DIALOG_SELECT
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapter.kt:247]: so if the questionnaire item type is not choice or reference, this extension will be no-op.

do you need to add this to any documentation?

**Review by jingtang10 — APPROVED** [review]
thank you!

File: datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapter.kt (hunk 4 above)
Comment (FikriMilano) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireEditAdapter.kt:247]: added

**Pull request merged**